service/frontend/dcRedirectionHandler.go

```diff
@@ -537,6 +537,9 @@ func (handler *DCRedirectionHandlerImpl) PollWorkflowTaskQueue(
 		return err
 	})
 
+	if resp == nil && err == nil {
+		return &workflowservice.PollWorkflowTaskQueueResponse{}, nil
```

Comment:
nit: maybe prefer `resp = &workflowservice.PollWorkflowTaskQueueResponse{}` over adding another function exit points especially for non-error cases?